src/tools/x/src/main.rs

```diff
@@ -0,0 +1,92 @@
+//! Run `x.py` from any subdirectory of a rust compiler checkout.
+//!
+//! We prefer `exec`, to avoid adding an extra process in the process tree.
+//! However, since `exec` isn't available on Windows, we indirect through
+//! `exec_or_status`, which will call `exec` on unix and `status` on Windows.
+//!
+//! We use `python`, `python3`, or `python2` as the python interpreter to run
+//! `x.py`, in that order of preference.
+
+use std::{
+    env, io,
+    process::{self, Command, ExitStatus},
+};
+
+const PYTHON: &str = "python";
+const PYTHON2: &str = "python2";
+const PYTHON3: &str = "python3";
+
+fn python() -> &'static str {
+    let val = match env::var_os("PATH") {
+        Some(val) => val,
+        None => return PYTHON,
+    };
+
+    let mut python2 = false;
+    let mut python3 = false;
+
+    for dir in env::split_paths(&val) {
+        if dir.join(PYTHON).exists() {
+            return PYTHON;
+        }
+
+        python2 |= dir.join(PYTHON2).exists();
+        python3 |= dir.join(PYTHON3).exists();
+    }
+
+    if python3 {
+        PYTHON3
+    } else if python2 {
+        PYTHON2
+    } else {
+        PYTHON
+    }
+}
+
+#[cfg(unix)]
+fn exec_or_status(command: &mut Command) -> io::Result<ExitStatus> {
+    use std::os::unix::process::CommandExt;
+    Err(command.exec())
+}
+
+#[cfg(not(unix))]
+fn exec_or_status(command: &mut Command) -> io::Result<ExitStatus> {
+    command.status()
+}
```

Comment:
Can we just always invoke status() here? I think the extra process shouldn't matter at all, right?

Reply:
We could invoke status, but if users are using something like `ps | grep` to see running processes, extra processes can get annoying, and make it confusing to know what to kill.

Reply:
Ok, resolving this -- I think this seems fine. We do this inside bootstrap I think already and it's not exactly lots of code either :)